src/annotator/test/sidebar-test.js

```diff
@@ -155,6 +155,27 @@ describe('Sidebar', () => {
     });
   });
 
+  describe('#ready', () => {
+    it('returns a promise that resolves when `hypothesisSidebarReady` message is received', async () => {
+      const sidebar = createSidebar();
+
+      // Check `sidebar.ready` is not resolved before `hypothesisSidebarReady`
+      // message is received.
+      assert.equal(
+        await Promise.race([sidebar.ready, Promise.resolve('not-ready')]),
+        'not-ready'
+      );
+
+      window.dispatchEvent(
+        new MessageEvent('message', {
+          data: { type: 'hypothesisSidebarReady' },
+        })
+      );
+
+      return sidebar.ready;
```

Comment:
I found this test a bit contrive. What about this alternative?
```suggestion
      const sidebar = createSidebar();

      window.dispatchEvent(
        new MessageEvent('message', {
          data: { type: 'hypothesisSidebarReady' },
        })
      );
      await sidebar.ready;

      assert(true);
```

Or maybe:

```suggestion
      const sidebar = createSidebar();
      let counter = 0;
      const promise = sidebar.ready.then(() => ++counter);

      window.dispatchEvent(
        new MessageEvent('message', {
          data: { type: 'hypothesisSidebarReady' },
        })
      );
      await promise;

      assert.equal(counter, 1);
```

Reply:
The `Promise.race` part is an attempt to check that the `sidebar.ready` Promise is not already resolved. JavaScript intentionally doesn't provide APIs to inspect Promise state, so the best we can do is race it against a Promise that is immediately resolved.

```
 const promise = sidebar.ready.then(() => ++counter);
 ```
 
Promises can only be resolved once, so unless `sidebar.ready` were re-assigned to a different Promise, `counter` can never have a value that is more than one here.